pkg/pub_integration/lib/src/fake_pub_server_process.dart

```diff
@@ -20,8 +20,13 @@ class FakePubServerProcess {
 
   FakePubServerProcess._(this.port, this._process);
 
-  static Future<FakePubServerProcess> start(
-      {String pkgDir, int port, int storagePort}) async {
+  static Future<FakePubServerProcess> start({
+    String pkgDir,
+    int port,
+    int storagePort,
+    bool collectCoverage = false,
+    int vmServicePort = 29999,
+  }) async {
     pkgDir ??= p.join(Directory.current.path, '../fake_pub_server');
     // TODO: check for free port
     port ??= 20000 + _random.nextInt(990);
```

Comment:
Unrelated to this PR, but usually you just use port zero and it picks a random ephemeral port.. Not sure if it always picks an idle port.

Reply:
but then we'd need to parse it to access it from the client. Shall I just use a fixed port?

Reply:
You'd get it back after listening, it's just a nit...